edx_lint/tamper_evident.py

```diff
@@ -50,19 +50,19 @@ def validate(self):
         with.
         """
 
-        with open(self.filename, "r") as f:
+        with open(self.filename, "rb") as f:
             text = f.read()
 
-        start_last_line = text.rfind("\n", 0, -1)
+        start_last_line = text.rfind(b"\n", 0, -1)
         if start_last_line == -1:
             return False
 
         original_text = text[:start_last_line+1]
         last_line = text[start_last_line+1:]
 
-        expected_hash = hashlib.sha1(original_text.encode("utf-8")).hexdigest()
-        match = re.search(r"[0-9a-f]{40}", last_line)
+        expected_hash = hashlib.sha1(original_text).hexdigest().encode('ascii')
```

Comment:
This encoding differs than the `actual_hash` encoding used above (not materially, I suppose, since the hexdigest is all ascii... but still, consistency...)